variables.tf

```diff
@@ -0,0 +1,162 @@
+    allow_basic_no_fast_forward   = optional(bool, false)
+    allow_rebase_with_merge       = optional(bool, false)
+  })
+  default = {
+    enabled                       = false
+    blocking                      = false
+    allow_squash                  = false
+    allow_rebase_and_fast_forward = false
+    allow_basic_no_fast_forward   = false
+    allow_rebase_with_merge       = false
+  }
+}
+
+variable "azuredevops_branch_policy_auto_reviewers" {
+  description = <<DESCRIPTION
+Auto reviewer policy settings.
+
+Fields:
+  - enabled (required): Whether the policy is enabled.
+  - blocking (required): Whether the policy is blocking.
+  - group_names (required): List of group names to assign as reviewers.
+  - submitter_can_vote (required): Whether the submitter can vote.
+  - message (required): Message to display when reviewers are assigned.
+  - minimum_number_of_reviewers (required): Minimum number of reviewers required.
+DESCRIPTION
+  type = object({
+    enabled                     = optional(bool, false)
+    blocking                    = optional(bool, false)
+    group_names                 = list(string)
+    submitter_can_vote          = optional(bool, false)
+    message                     = optional(string, "Code Reviewers")
+    minimum_number_of_reviewers = optional(number, 1)
+  })
+  default = {
+    enabled                     = false
+    blocking                    = false
+    group_names                 = []
+    submitter_can_vote          = false
+    message                     = "Code Reviewers have been automatically assigned to this pull request."
+    minimum_number_of_reviewers = 1
+  }
+}
+
+variable "azuredevops_branch_policy_build_validation" {
+  description = <<DESCRIPTION
+Branch policy build validation settings.
+
+Fields:
+  - enabled (optional): Whether the policy is enabled. Default is false.
+  - blocking (optional): Whether the policy is blocking. Default is false.
+  - valid_duration (optional): Duration (in minutes) for which the build is valid. Default is 720.
+  - manual_queue_only (optional): Whether the build must be manually queued. Default is false.
+  - queue_on_source_update_only (optional): Queue builds only on source updates. Default is true.
+  - suffix (optional): Suffix to append to the build policy name. Default is an empty string.
+DESCRIPTION
+  type = object({
+    enabled                     = optional(bool, false)
+    blocking                    = optional(bool, false)
+    valid_duration              = optional(number, 720)
+    manual_queue_only           = optional(bool, false)
+    queue_on_source_update_only = optional(bool, true)
+    suffix                      = optional(string, "")
+  })
+  default = {
+    enabled                     = false
+    blocking                    = false
+    valid_duration              = 720
+    manual_queue_only           = false
+    queue_on_source_update_only = true
+    suffix                      = ""
+  }
+}
```

Comment:
Any outputs we might want to export from this module? Maybe the list of projects we are querying or the actual settings objects for each project?

Might not be useful now but could be good to have later.